### ISSUE TITLE
Bad interaction between 'dune install' and 'opam' in restricted file system

### ISSUE THREAD

**Issue by mjambon** [ocaml/dune#5699]
## Expected Behavior

`opam install easy-format.1.3.3` should work. This is an ordinary, pure ocaml library, whose opam package generated from `dune-project` recently changed and now includes a `dune install` command.

## Actual Behavior

`opam install easy-format.1.3.3` fails for users whose opam root is not standard, with the message:
```
[ERROR] The compilation of easy-format.1.3.3 failed at "dune install -p easy-format --create-install-files easy-format".
<snip>
# [ERROR] /media/mand/.opam exists, but does not appear to be a valid opam root. Please remove it and use `opam init', or specify a different `--root' argument
```
See full report here: https://github.com/ocaml-community/easy-format/issues/28

## Reproduction

See https://github.com/ocaml-community/easy-format/issues/28

## Specifications

```
➜   opam --version
2.1.2
➜   dune --version
3.1.1
```

@eponier wrote:
> Note that the future opam 2.2 solves the problem completely since nearly all the filesystem will be available by default.

## Additional information 

The issue is explained in detail by @eponier [here](https://github.com/ocaml-community/easy-format/issues/28#issuecomment-1125153476).

Somehow, we need to make sure all the opam packages that get released in opam-repository will install properly with the latest version of dune and opam < 2.2 for users with a nonstandard opam root.

**Comment by rgrinberg** [comment]
Should be fixed in the upcoming release: 3.2.0

**Comment by mjambon** [comment]
many thanks, @rgrinberg!
